### ISSUE TITLE
When checking Pydantic derived models mypy ignores default values of parent fields

### ISSUE THREAD

**Issue by stur86** [python/mypy#14850]
**Bug Report**

When using Pydantic models, if one creates a parent model with fields with default values, then a child model inheriting from it, and instantiates the child model, mypy erroneously complains for the lack of named arguments corresponding to the parent model's fields.

**To Reproduce**

Simplest code:

```python
from pydantic import BaseModel, Field

class ParentModel(BaseModel):
    a: str = Field("A")


class ChildModel(ParentModel):
    b: str = Field("B")


if __name__ == "__main__":
    c: ChildModel = ChildModel(b="X")
```

**Expected Behavior**

If the field has a default value, it's not necessary to pass it, and it should not be considered an error to not do so.

**Actual Behavior**

The named argument is mentioned as missing even though it has a default value:

```
mypy_pydantic_bug.py:12: error: Missing named argument "a" for "ChildModel"  [call-arg]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.1.1
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.4

**Comment by chassing** [comment]
And it doesn't honor `allow_population_by_field_name` config anymore.

[allow_population_by_field_name](https://docs.pydantic.dev/usage/model_config/#options): 
whether an aliased field may be populated by its name as given by the model attribute, as well as the alias (default: False)

**Comment by erictraut** [comment]
If you use a keyword argument (`default=`) for the field, it works.

```python
class ParentModel(BaseModel):
    a: str = Field(default="A")

class ChildModel(ParentModel):
    b: str = Field(default="B")
```

Pydantic uses `dataclass_transform` (defined in [PEP 681](https://peps.python.org/pep-0681/)), which mypy now supports. The `dataclass_transform` mechanism requires you to specify field arguments by keyword because this is how the stdlib `dataclass` works.

**Comment by wesleywright** [comment]
To elaborate a bit on the comments above: unfortunately some Pydantic features aren't fully covered by the PEP 681 specification, among them the defaults and `allow_population_by_field_name`. Either you'll need to use the custom plugin that Pydantic provides or you'll have to avoid using these features of Pydantic.

**Comment by erictraut** [comment]
Summary: If you stick with the standard features of `dataclass_transform`, mypy is able to type check the code without any plugins. If you use more bespoke features of pydantic, then you need to install the pydantic plugin for mypy.

Is there anything actionable remaining in this bug report?

**Comment by irdkwmnsb** [comment]
I've spent too much time looking for an answer of my issue after upgrading to pydantic v2.
Since this is the first google result for the query of "pydantic mypy Missing named argument", I feel obligated to add this:
### Possible resolvement:
If you are using Optional[type] for any of your fields, make sure you add Field(default=None), as per [documentation](https://docs.pydantic.dev/latest/concepts/fields/#default-values)
Also see [pydantic/6713](https://github.com/pydantic/pydantic/issues/6713)

**Comment by gouravsinghbais** [comment]
> If you use a keyword argument (`default=`) for the field, it works.
> 
> ```python
> class ParentModel(BaseModel):
>     a: str = Field(default="A")
> 
> class ChildModel(ParentModel):
>     b: str = Field(default="B")
> ```
> 
> Pydantic uses `dataclass_transform` (defined in [PEP 681](https://peps.python.org/pep-0681/)), which mypy now supports. The `dataclass_transform` mechanism requires you to specify field arguments by keyword because this is how the stdlib `dataclass` works.

This one works pretty well for my case, thanks.
